src/common/libjob/job_info.h

```diff
@@ -0,0 +1,36 @@
+/************************************************************\
+ * Copyright 2018 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#ifndef _FLUX_CORE_JOB_EVENTLOG_H
+#define _FLUX_CORE_JOB_EVENTLOG_H
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <flux/core.h>
+
+#include "job_types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+flux_future_t *flux_job_eventlog_lookup (flux_t *h, int flags, flux_jobid_t id);
+int flux_job_eventlog_lookup_get (flux_future_t *f, const char **event);
```

Comment:
I wonder if we could actually drop the flags and rename to `flux_job_event_watch()`?  I was plannig to add a `flux_job_state_watch()` that would return one state per response, so this could be kind of paired with that?

We'll need a function for fetching static job info in bulk (R, jobspec, J, etc).  Maybe that one could be used to fetch the eventlog in its entirety?

I'm not at all sure what the right interfaces are here, so one possibility is to go with what you have and tweak later.

Reply:
Agreed, I think exported interfaces in `libjob` related to job "events" could hide the fact from (most) users that the events are in an event log. I like the parity with `flux_job_state_watch`.

I'm also fine starting with what you have though. Probably more important are the cmdline utilities and Python interface...

Reply:
So are you thinking `flux_job_event_watch()`, `flux_job_event_watch_get()`, and `flux_job_event_watch_cancel()`?  Sounds like a good idea if there will be other similar ones.

> We'll need a function for fetching static job info in bulk (R, jobspec, J, etc). Maybe that one could be used to fetch the eventlog in its entirety?

Yeah, once that API / interface becomes more clear, I think that could easily be used for the full eventlog retrieval.